frontend/public/components/routes/utils.ts

```diff
@@ -0,0 +1,144 @@
+import * as _ from 'lodash-es';
+import * as yup from 'yup';
+import { K8sResourceKind } from '@console/internal/module/k8s';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { getActiveNamespace } from '../../actions/ui';
+import { AlternateServiceEntryType, RouteFormProps, UNNAMED_PORT_KEY } from './create-route';
+
+export const routeValidationSchema = yup.object({
+  editorType: yup.string(),
+  formData: yup.object().when('editorType', {
+    is: EditorType.Form,
+    then: yup.object({
+      name: yup.string().required(),
+      service: yup.string().required(),
+      targetPort: yup.string().required(),
+      termination: yup.string().when('formData.secure', {
+        is: true,
+        then: yup.string().required(),
+      }),
+    }),
+  }),
+});
+
+export const convertRouteToEditForm = (
+  services: K8sResourceKind[],
+  data: K8sResourceKind,
+): Partial<RouteFormProps> => {
+  if (!data) {
+    return null;
+  }
+  const { metadata, spec } = data;
+  return {
+    name: metadata?.name,
+    hostname: spec?.host,
+    path: spec?.path,
+    service: services?.find((s) => s.metadata.name === spec?.to?.name),
+    weight: spec?.to?.weight,
+    targetPort: spec?.port.targetPort,
```

Comment:
Add optional chaining as `spec.port` is an [optional field](https://github.com/openshift/console/blob/master/frontend/public/module/k8s/types.ts#L583-L585) 
```suggestion
    targetPort: spec?.port?.targetPort,
```
UI is breaking if I visit edit Route form for the routes without port (created via yaml view). 
```
TypeError
Cannot read properties of undefined (reading 'targetPort')
```

Reply:
Updated.

Reply:
Did not know that target port was optional, @jerolimov should we remove the required flag for this field? This would change the form behavior.